packages/tools/src/tools/ScaleOverlayTool.ts

```diff
@@ -0,0 +1,666 @@
+import AnnotationDisplayTool from './base/AnnotationDisplayTool';
+import { vec3 } from 'gl-matrix';
+import {
+  metaData,
+  getRenderingEngines,
+  utilities as csUtils,
+} from '@cornerstonejs/core';
+import { ScaleOverlayAnnotation } from '../types/ToolSpecificAnnotationTypes';
+import type { Types } from '@cornerstonejs/core';
+import { filterViewportsWithToolEnabled } from '../utilities/viewportFilters';
+import { addAnnotation } from '../stateManagement/annotation/annotationState';
+import {
+  drawLine as drawLineSvg,
+  drawTextBox as drawTextBoxSvg,
+} from '../drawingSvg';
+import {
+  EventTypes,
+  PublicToolProps,
+  ToolProps,
+  SVGDrawingHelper,
+} from '../types';
+import { StyleSpecifier } from '../types/AnnotationStyle';
+
+const SCALEOVERLAYTOOL_ID = 'scaleoverlay-viewport';
+
+/**
+ * @public
+ * @class ScaleOverlayTool
+ * @memberof Tools
+ *
+ * @classdesc Tool for displaying a scale overlay on the image.
+ * @extends Tools.Base.BaseTool
+ */
+class ScaleOverlayTool extends AnnotationDisplayTool {
+  static toolName;
+
+  public touchDragCallback: any;
+  public mouseDragCallback: any;
+  _throttledCalculateCachedStats: any;
+  editData: {
+    renderingEngine: any;
+    viewport: any;
+    annotation: ScaleOverlayAnnotation;
+  } | null = {} as any;
+  isDrawing: boolean;
+  isHandleOutsideImage: boolean;
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      configuration: {
+        viewportId: '',
+        scaleLocation: 'bottom',
+        scaleColor: 'yellow',
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+  }
+
+  _init = (): void => {
+    const renderingEngines = getRenderingEngines();
+    const renderingEngine = renderingEngines[0];
+
+    if (!renderingEngine) {
+      return;
+    }
+
+    let viewports = renderingEngine.getViewports();
+    viewports = filterViewportsWithToolEnabled(viewports, this.getToolName());
```

Comment:
instead of this you should do 

`const viewportIds = getToolGroup(this.toolGroupId).viewportsInfo;` 

you can grab the viewport from the toolGroupId, see the refernceCursors implementation
packages/tools/src/tools/ReferenceCursors.ts